Sources/SwiftDocC/Infrastructure/External Data/OutOfProcessReferenceResolver.swift

```diff
@@ -762,30 +810,54 @@ extension OutOfProcessReferenceResolver {
                     willEventuallyBeDeprecated: false // This information isn't used anywhere since this node doesn't have its own page, it's just referenced from other pages.
                 )
             })
-        } else {
-            availability = nil
+            availabilityVariants[mainTrait] = availability
+        }
+        
+        var kindVariants = [mainTrait: symbolKind(forNodeKind: kind)]
+        var titleVariants = [mainTrait: title]
+        var subHeadingVariants = [DocumentationDataVariantsTrait: [SymbolGraph.Symbol.DeclarationFragments.Fragment]]()
+        subHeadingVariants[mainTrait] = declarationFragments?.declarationFragments
+        var abstractVariants = [DocumentationDataVariantsTrait: AbstractSection]()
+        abstractVariants[mainTrait] = abstract.map { AbstractSection(paragraph: Paragraph([Text($0)])) }
+        
+        for variant in variants ?? [] {
+            guard case .interfaceLanguage(let language) = variant.traits.first else { continue }
+            let trait = DocumentationDataVariantsTrait(interfaceLanguage: language)
+            
+            if let variantKind = variant.kind {
+                kindVariants[trait] = symbolKind(forNodeKind: variantKind)
+            }
+            if let variantTitle = variant.title {
+                titleVariants[trait] = variantTitle
+            }
+            if let variantAbstract = variant.abstract {
+                abstractVariants[trait] = AbstractSection(paragraph: Paragraph([Text(variantAbstract)]))
+            }
+            if let variantDeclarationFragments = variant.declarationFragments {
+                subHeadingVariants[trait] = variantDeclarationFragments?.declarationFragments
```

Comment:
```suggestion
            if let variantDeclarationFragments = variant.declarationFragments?.declarationFragments {
                subHeadingVariants[trait] = variantDeclarationFragments
```

Not sure if this will be an issue here- but I ran into bugs in other places working with variants where I was accidentally setting a variant field to `nil` which would cause it to be explicitly removed when really the behavior I want is to fall back to the other variant.

Reply:
I added a comment here to describe the optional behavior

Reply:
I'm not seeing the comment. Did it get committed?

Reply:
:facepalm: For some reason I had staged these changes thinking that I had committed them. I've pushed two new commits now.